cpp/src/gandiva/engine.cc

```diff
@@ -268,49 +283,104 @@ Status Engine::LoadPreCompiledIR() {
 // a pass for dead code elimination.
 Status Engine::RemoveUnusedFunctions() {
   // Setup an optimiser pipeline
-  std::unique_ptr<llvm::legacy::PassManager> pass_manager(
-      new llvm::legacy::PassManager());
+  llvm::PassBuilder pass_builder;
+  llvm::ModuleAnalysisManager module_am;
+
+  pass_builder.registerModuleAnalyses(module_am);
+  llvm::ModulePassManager module_pm;
 
   std::unordered_set<std::string> used_functions;
   used_functions.insert(functions_to_compile_.begin(), functions_to_compile_.end());
 
-  pass_manager->add(
-      llvm::createInternalizePass([&used_functions](const llvm::GlobalValue& func) {
-        return (used_functions.find(func.getName().str()) != used_functions.end());
+  module_pm.addPass(
+      llvm::InternalizePass([&used_functions](const llvm::GlobalValue& GV) -> bool {
+        return used_functions.find(GV.getName().str()) != used_functions.end();
```

Comment:
How about lower case for like others?

```suggestion
      llvm::InternalizePass([&used_functions](const llvm::GlobalValue& gv) -> bool {
        return used_functions.find(gv.getName().str()) != used_functions.end();
```

Or can we use `variable` instead of `gv` for readability?

Reply:
I'll use `variable` and merge this to fix CI failures for now:

```suggestion
      llvm::InternalizePass([&used_functions](const llvm::GlobalValue& variable) -> bool {
        return used_functions.find(variable.getName().str()) != used_functions.end();
```

We can improve this later in a follow-up PR if it's need.